app/lib/grades/pages/grades_dialog/fields.dart

```diff
@@ -0,0 +1,553 @@
+                  GradingSystem.zeroToFivteenPointsWithDecimals => 'z.B. 15.0',
+                  GradingSystem.zeroToHundredPercentWithDecimals => 'z.B. 78.8',
+                  _ => null,
+                },
+                errorText: view.selectedGradeErrorText,
+              ),
+              onChanged: controller.setGrade,
+              keyboardType:
+                  const TextInputType.numberWithOptions(decimal: true),
+              inputFormatters: <TextInputFormatter>[
+                FilteringTextInputFormatter.allow(RegExp('[0-9.,]'))
+              ],
+            ),
+      onTap: hasDistinctValues
+          ? () async {
+              final res = await showDialog<String?>(
+                context: context,
+                builder: (context) => SimpleDialog(
+                  title: const Text("Note auswählen"),
+                  contentPadding: const EdgeInsets.all(12),
+                  children: [
+                    if (selectableGrades.distinctGrades != null)
+                      for (final grade in selectableGrades.distinctGrades!)
+                        ListTile(
+                          title: Text(grade),
+                          onTap: () {
+                            Navigator.of(context).pop<String?>(grade);
+                          },
+                        ),
+                  ],
+                ),
+              );
+
+              if (res != null) {
+                controller.setGrade(res);
+              }
+            }
+          : null,
+    );
+  }
+}
+
+class _GradingSystem extends StatelessWidget {
+  const _GradingSystem();
+
+  @override
+  Widget build(BuildContext context) {
+    final controller = Provider.of<GradesDialogController>(context);
+    final view = controller.view;
+    return ListTile(
+      leading: SavedGradeIcons.gradingSystem,
+      title: const Text("Notensystem"),
+      subtitle: Text(view.selectedGradingSystem.displayName),
+      onTap: () async {
+        final res = await showDialog<GradingSystem?>(
+          context: context,
+          builder: (context) => const _SelectGradeSystemDialog(),
+        );
+
+        if (res != null) {
+          controller.setGradingSystem(res);
+        }
+      },
+    );
+  }
+}
+
+class _SelectGradeSystemDialog extends StatelessWidget {
+  const _SelectGradeSystemDialog();
+
+  @override
+  Widget build(BuildContext context) {
+    final greyColor = Theme.of(context).colorScheme.onSurface.withOpacity(0.5);
+    return MaxWidthConstraintBox(
+      maxWidth: 450,
+      child: SimpleDialog(
+        title: const Text("Notensystem auswählen"),
+        contentPadding: const EdgeInsets.fromLTRB(12, 12, 12, 16),
+        children: [
+          Padding(
+            padding: const EdgeInsets.fromLTRB(12, 0, 12, 12),
+            child: Text(
+              'Der erste Wert entspricht der besten Noten, z.B. bei dem Notensystem "1 - 6" ist "1" die beste Note.',
```

Comment:
This isn't something that affects the calculations but will be important for the grading performance